sql/core/src/test/scala/org/apache/spark/status/protobuf/sql/KVStoreProtobufSerializerSuite.scala

```diff
@@ -179,28 +146,33 @@ class KVStoreProtobufSerializerSuite extends SparkFunSuite {
     assert(result.nodes.size == input.nodes.size)
 
     def compareNodes(n1: SparkPlanGraphNodeWrapper, n2: SparkPlanGraphNodeWrapper): Unit = {
-      assert(n1.node.id == n2.node.id)
-      assert(n1.node.name == n2.node.name)
-      assert(n1.node.desc == n2.node.desc)
-
-      assert(n1.node.metrics.size == n2.node.metrics.size)
-      n1.node.metrics.zip(n2.node.metrics).foreach { case (m1, m2) =>
-        assert(m1.name == m2.name)
-        assert(m1.accumulatorId == m2.accumulatorId)
-        assert(m1.metricType == m2.metricType)
-      }
-
-      assert(n1.cluster.id == n2.cluster.id)
-      assert(n1.cluster.name == n2.cluster.name)
-      assert(n1.cluster.desc == n2.cluster.desc)
-      assert(n1.cluster.nodes.size == n2.cluster.nodes.size)
-      n1.cluster.nodes.zip(n2.cluster.nodes).foreach { case (n3, n4) =>
-        compareNodes(n3, n4)
-      }
-      n1.cluster.metrics.zip(n2.cluster.metrics).foreach { case (m1, m2) =>
-        assert(m1.name == m2.name)
-        assert(m1.accumulatorId == m2.accumulatorId)
-        assert(m1.metricType == m2.metricType)
+      if (n1.node != null) {
```

Comment:
Should we add an additional check related to `toSparkPlanGraphNode` in this suite due to the special scenario

Reply:
Updated